### DISCUSSION TITLE
How to copy the changes from one branch to a sibling branch

### DISCUSSION THREAD

**Discussion by guostong** [community/community#143466] [Repositories] [answered]
### Select Topic Area

Question

### Body

Our version control structure is as below:
Dev and production release branch are sibling, they are all branched out from master, when we apply hotfix production release branch, we will manually apply the same changes to dev branch. My question is can we somehow copy the changes from Production release branch to Dev branch?

**Comment by quintyn** [comment, marked as answer]
Yes to copy changes from prod release to dev, you can:

1.	Merge: Switch to the Dev branch and merge in the Production branch. This brings all changes over at once.

git checkout dev
git merge production-release-branch


2.	Cherry-pick: If you only want specific commits (like a hotfix), you can cherry-pick them from Production into Dev.

git checkout dev
git cherry-pick <commit-hash>


3.	Rebase: For a cleaner history, you could rebase Dev onto the Production branch, though this can be more complex if there are conflicts.

git checkout dev
git rebase production-release-branch

**Reply by guostong** [reply]
thank you, cherry-pick works

**Comment by radanpro** [comment]
There are several ways to achieve this.
 I will assume that you want to merge via `GitHub`,  but if you prefer to merge using `Git` on your machine, feel free to let me know.

## Method via GitHub
Let’s assume you have two sibling branches named **Production** and **Dev**, both branched out from the main branch **Master**, and you want to merge changes between these two branches without affecting **Master**. Here are the steps to accomplish this:

1. **Create a Pull Request between the sibling branches**:
   - Open your project in GitHub and navigate to the **Pull Requests** tab.
   - Click on **New Pull Request** to create a new merge request.
   - Select the **Dev** branch as the base branch that will receive the changes, and the **Production** branch as the compare branch from which to pull changes.

2. **Review and confirm the request**:
   - After selecting the branches, you will see a comparison of the proposed changes. Be sure to review the changes to ensure they are correct and align with what you want to merge.
   - Click on **Create Pull Request** to initiate the request, and you can add a description explaining the nature of the changes if you wish.

3. **Merge the changes without affecting the main branch**:
   - After creating the **Pull Request**, you can review the request and approve it by clicking **Merge**.
   - This way, the changes will only be merged between **Production** and **Dev**, without transferring any of these changes to **Master**.

> **Note**: As long as the **Pull Request** was created between the sibling branches **Production** and **Dev** only, it will not affect the main branch **Master**.

**Reply by guostong** [reply]
Thank you, sounds like the solution I expected, it will be great if you can provide a git command solution

**Reply by radanpro** [reply]
Certainly, the solution using Git is much simpler. Here are the steps:

Assuming you have two sibling branches (as explained in the GitHub method), **Production** and **Dev**, and you want to merge changes from **Production** into **Dev** without affecting **Master**.

1. **Switch to the target branch**:
   ```bash
   git checkout Dev
   ```

2. **Merge changes from the source branch**:
   ```bash
   git merge Production
   ```

3. **Push changes to GitHub (optional)**:
   ```bash
   git push origin Dev
   ```

> **Note**: This approach keeps the changes confined to **Production** and **Dev** only, without impacting **Master**.

**Comment by davevad93** [comment]
Hi @guostong ,

You can merge the changes from `Production release` to `Dev` with `git` commands by doing this:

1.  Switch to `Dev` branch.
```bash
git checkout Dev
```
2. Merge the commits from `Production release` to the current branch which is `Dev`.
```bash
git merge Production release
```
3. Push the changes to the remote repo.
```bash
git push origin Dev
```
